src/python/dxpy/bindings/dxfile_functions.py

```diff
@@ -96,17 +96,19 @@ def _get_executor():
         _executor = ThreadPoolExecutor(max_workers=cpu_count())
     return _executor
 
-def download_dxfile(dxfile_or_id, filename, chunksize=dxfile.DEFAULT_BUFFER_SIZE, append=False, show_progress=False,
-                    project=None, **kwargs):
+def download_dxfile(dxid, filename, dxfile=None, chunksize=dxfile.DEFAULT_BUFFER_SIZE, append=False,
+                    show_progress=False, project=None, **kwargs):
```

Comment:
If you really want to guarantee backwards compatibility, retain the same ordering of all the arguments that existed previously and add your new one at the end:

```
def download_dxfile(dxid, filename, chunksize=dxfile.DEFAULT_BUFFER_SIZE, append=False, show_progress=False,
                    project=None, dxfile=None, **kwargs):
``

However, both this and the proposed API above are kind of annoying to use. `dxid` and `dxfile` are mutually exclusive, but if you supply `dxfile` as a keyword argument you then still have to override `dxid=None`. One option is to make all the arguments keyword:

```
def download_dxfile(dxid=None, filename=None, ..., dxfile=None, **kwargs):
```

But I find this kind of unsatisfying too since you lose the statically available information that at least two arguments are needed. Frankly I'd be fine with restoring the previous signature exactly and just documenting the fact that you can supply a DXFile or an ID in the `dxid` position. That might be the least bad option, actually.